src/domain/human-description/entities/human-description-template.entity.ts

```diff
@@ -46,9 +55,8 @@ export class HumanDescriptionTemplate {
     });
 
     const fragments: HumanDescriptionFragment[] = [];
-    const matches = this.template.matchAll(HumanDescriptionTemplate.REGEX);
 
-    for (const match of matches) {
+    for (const match of this.templateMatches) {
       if (!match.groups) throw Error(`Error parsing template ${this.template}`);
```

Comment:
This error can be moved to the constructor since we do the `matchAll` there. At that point we already know if this template is valid or not.

Reply:
Makes sense! I've moved the check into the constructor.